### PR TITLE
tunnel: use conn.BindSocketToInterface type

### PR DESCRIPTION
**Pull request by crawshaw**
Built and run on windows. I still see `Binding v4 socket to interface` with this commit.

### CODE REVIEW

File: tunnel/defaultroutemonitor.go [hunk 1]
```diff
@@ -51,12 +51,14 @@ func bindSocketRoute(family winipcfg.AddressFamily, device *device.Device, ourLU
 	*lastLUID = luid
 	*lastIndex = index
 	blackhole := blackholeWhenLoop && index == 0
-	if family == windows.AF_INET {
-		log.Printf("Binding v4 socket to interface %d (blackhole=%v)", index, blackhole)
-		return device.BindSocketToInterface4(index, blackhole)
-	} else if family == windows.AF_INET6 {
-		log.Printf("Binding v6 socket to interface %d (blackhole=%v)", index, blackhole)
-		return device.BindSocketToInterface6(index, blackhole)
+	if bind, _ := device.Bind().(conn.BindSocketToInterface); bind != nil {
```
Comment (zx2c4) [inline tunnel/defaultroutemonitor.go:54]: This should return an error if it fails.

Comment (crawshaw) [reply tunnel/defaultroutemonitor.go]: Done.

**Comment by zx2c4** [comment]
I'm still waiting to see the corresponding wireguard-go commit and to see this merged. What's going on @crawshaw ?

**Comment by crawshaw** [comment]
Ah I was going to get this in then just delete the shim file, as this can be applied now. I'll send a wireguard-go PR for the deletion tomorrow.

**Comment by crawshaw** [comment]
The followup wireguard-go change is https://github.com/WireGuard/wireguard-go/pull/35

**Pull request closed without merging**